src/eherkenning/backends.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth import get_user_model
+
+from digid_eherkenning.backends import eHerkenningBackend as _eHerkenningBackend
+from digid_eherkenning.exceptions import eHerkenningError
+from digid_eherkenning.utils import get_client_ip
+
+UserModel = get_user_model()
+
+
+class eHerkenningBackend(_eHerkenningBackend):
+    """
+    Custom backend to identify users based on the KvK number instead of RSIN
+    """
+
+    def get_or_create_user(self, request, saml_response, saml_attributes):
+        kvk = self.get_kvk_number(saml_attributes)
+        if kvk == "":
+            raise eHerkenningError(
+                "Login failed due to no KvK being returned by eHerkenning."
+            )
+
+        created = False
+        try:
+            user = UserModel.eherkenning_objects.get_by_kvk(kvk)
```

Comment:
zou nog fout kunnen gaan bij meerdere users met hetzelfde kvk nummer. Maar dat is nu ook het geval voor DigiD en een 500 in deze situatie lijkt mij prima.